### PR TITLE
feat: adds debug_traceBlockByNumber.callFlatTracer

### PR DESCRIPTION
**Pull request by HermanObst**
## What ❔

This PR adds support for ‘Parity style’ output for `debug_traceBlockByNumber` tracer. 
Specifically adds `.callFlatTracer` to modify the format of the tracer output so that it’s the same format as Parity’s tracer output.

## Why ❔

We want to provide a way to get the the same format as parity in `debug_traceBlockByNumber`

## Checklist

- [X] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [X] Tests for the changes have been added / updated.
- [X] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.
- [X] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Review by Deniallugo — COMMENTED** [review]
Looks good! 
I'd like to move it outside mod.rs and add more tests, but generally the approach is correct

**Review by Deniallugo — COMMENTED** [review]
Could you please add a test, where we use the new functionality only? 
I mean where you convert one struct to another, it will be really helpful and more visible

**Comment by HermanObst** [comment]
> Could you please add a test, where we use the new functionality only? I mean where you convert one struct to another, it will be really helpful and more visible

Sure, I'll tackle this today!

File: core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs [hunk 1]
```diff
@@ -79,7 +80,7 @@ impl DebugNamespace {
             .get_traces_for_miniblock(block_number)
             .await
             .context("get_traces_for_miniblock")?;
-        let call_trace = call_traces
+        let call_trace: Vec<ResultDebugCall> = call_traces
```
Comment (Deniallugo) [inline core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs:83]: nit: seems you don't need to specify the type.

File: core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs [hunk 2]
```diff
@@ -92,6 +93,18 @@ impl DebugNamespace {
...
 
+    #[tracing::instrument(skip(self))]
+    pub async fn debug_trace_block_flat_impl(
+        &self,
+        block_id: BlockId,
+        options: Option<TracerConfig>,
+    ) -> Result<Vec<DebugCallFlat>, Web3Error> {
+        let call_trace = self.debug_trace_block_impl(block_id, options).await?;
+        let call_trace_flat = flatten_debug_calls(call_trace);
+        println!("{:?}", call_trace_flat);
```
Comment (Deniallugo) [inline core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs:104]: seems it's a debug leftover

**Review by Deniallugo — COMMENTED** [review]
LGTM

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**